library/arith.lisp

```diff
@@ -480,6 +480,99 @@ The fields are defined as follows:
       (lisp Double-Float (x y)
         (cl:coerce (cl:/ x y) 'cl:double-float)))))
 
+(coalton-toplevel
+  (define-class (Trigonometric :a)
+    (sin   (:a -> :a))
+    (cos   (:a -> :a))
+    (tan   (:a -> :a))
+    (sinh  (:a -> :a))
+    (cosh  (:a -> :a))
+    (tanh  (:a -> :a))
+    (asin  (:a -> :a))
+    (acos  (:a -> :a))
+    (atan  (:a -> :a))
+    (asinh (:a -> :a))
+    (acosh (:a -> :a))
+    (atanh (:a -> :a)))
```

Comment:
I would also include `atan2` and `sincos`

Reply:
done